server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/compaction/files/InactiveSystemCompFiles.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.tserver.tablet.compaction.files;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.accumulo.core.metadata.StoredTabletFile;
+import org.apache.accumulo.core.spi.compaction.CompactionKind;
+
+// NOT ACTIVE or CANCELLED
+public class InactiveSystemCompFiles implements CompactionFiles {
+
+  private final Set<StoredTabletFile> allCompactingFiles;
+
+  public InactiveSystemCompFiles(Set<StoredTabletFile> allCompactingFiles) {
+    this.allCompactingFiles = allCompactingFiles;
+  }
+
+  @Override
+  public Set<StoredTabletFile> getCandidates(Set<StoredTabletFile> currFiles, CompactionKind kind) {
+    Set<StoredTabletFile> candidates = new HashSet<>(currFiles);
+    candidates.removeAll(allCompactingFiles);
+    return Collections.unmodifiableSet(candidates);
```

Comment:
```suggestion
    return Collections.unmodifiableSet(Sets.difference(currFiles, allCompactingFiles));
```
This might be a nice change.

Reply:
I think this could be even simpler. It looks like Sets.difference() "Returns an unmodifiable view of the difference of two sets"

Reply:
I do wonder why the `currFiles` set was wrapped in a new HashSet. I think the type can be more specific to avoid wrapping.